### PR TITLE
Fix Slow DiskThresholdSettings

### PR DESCRIPTION
**Pull request by original-brownbear**
This is only really relevant to test execution but it takes many seconds
to parse these settings during a run of the server:internalClusterTests for example
due to constant throwing in the adjusted methods. Added some trivial short-circuits
and slightly optimized `ByteSizeValue` for some defaults and to make it so this
class effectively disappears from profiling.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java [hunk 1]
```diff
@@ -182,11 +182,13 @@ public void validate(final String value, final Map<Setting<?>, Object> settings)
     }
 
     private static void doValidate(String low, String high, String flood) {
-        try {
-            doValidateAsPercentage(low, high, flood);
-            return; // early return so that we do not try to parse as bytes
-        } catch (final ElasticsearchParseException e) {
-            // swallow as we are now going to try to parse as bytes
+        if (low.endsWith("b") == false) { // only try to validate as percentage if it isn't obviously a byte size value
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java:185]: Seems a bit odd just to take the shortcut on a lower-case `b` although I see that it helps. Let's extract the predicate to a method `definitelyNotPercentage(String)` so we can add a slightly longer comment about why we do this particular check.

**Review by DaveCTurner — COMMENTED** [review]
Seems like a good approach, I left one ask.

File: server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java [hunk 2]
```diff
@@ -202,6 +204,19 @@ public static ByteSizeValue parseBytesSizeValue(String sValue, ByteSizeValue def
...
             return defaultValue;
         }
+        switch (sValue) {
+            case "0":
+            case "0b":
+            case "0B":
+                return ZERO;
+            case "1b":
+            case "1B":
+                return ONE;
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java:214]: This is correct but looks like an accident, suggest a comment here too.

```suggestion
                // "1" is deliberately omitted, the units are required for all values except "0" and "-1"
                return ONE;
```

**Review by DaveCTurner — APPROVED** [review]
LGTM (one optional suggestion)

**Comment by original-brownbear** [comment]
Thanks David!

**Pull request merged**